lib/metrics.ts

```diff
@@ -0,0 +1,42 @@
+import * as _ from 'lodash';
+
+interface Metrics {
+  fetchCallGraphBuilder?: number;
+  getMvnClassPath?: number;
+  getGradleClassPath?: number;
+  getEntrypoints?: number;
+  generateCallGraph?: number;
+  mapClassesPerJar?: number;
+  getCallGraph?: number;
+}
+
+const metricsState: { [x in keyof Metrics]: [number, number] | undefined } = {};
```

Comment:
Maybe nicer to have it:
```
{[metric-name]: {start: num, end: num}}
```
I think it's more descriptive

Reply:
the `[number, number]` is the output from hrtime, it doesn't have start and end, just elapsed time.